realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -897,24 +897,139 @@ JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeNotEqual__J_3JLja
     TableQuery_StringPredicate(env, nativeQueryPtr, columnIndexes, value, caseSensitive, StringNotEqual);
 }
 
-JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeBeginsWith(
+JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeBeginsWith__J_3JLjava_lang_String_2Z(
```

Comment:
Also, I should note that according to [Oracle's `javah` documentation](http://docs.oracle.com/javase/8/docs/technotes/tools/windows/javah.html) the `-jni` option is the default behavior, so actually having it explicitly specified is a no-op anyway